backend/game/dice_handler.py

```diff
@@ -0,0 +1,27 @@
+import game.dice.dice as dice
+
+
+class DiceHandler:
+    def __init__(self):
+        self.dice_values = []
+        self.re_rolls_left = 0
+
+    def start_turn(self, starting_dice_count, re_roll_count):
+        self.dice_values = dice.roll_many(starting_dice_count)
+        self.re_rolls_left = re_roll_count
+
+    def re_roll_dice(self, indexes_of_dice_to_re_roll):
+        if self.re_rolls_left <= 0:
```

Comment:
I can see this if statement being it's own function as later on things may change the amount of rerolls left.  If not wanting to make it it's own function that returns a boolean, you can just change the logic to
`if self.re_rolls_left > 0`
that way you do not have to have an awkward return in the middle which kind of muddles the purpose of the function

Reply:
I like the idea of flipping the logic for legibility.

Reply:
applied change